File: NayanTv/app/src/main/java/com/iptv/nayantv/PlayerActivity.java
package com.iptv.nayantv;

import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.bumptech.glide.Glide;
import com.google.android.exoplayer2.ExoPlayer;
import com.google.android.exoplayer2.MediaItem;
import com.google.android.exoplayer2.Player;
import com.google.android.exoplayer2.source.hls.HlsMediaSource;
import com.google.android.exoplayer2.ui.StyledPlayerView;
import com.google.android.exoplayer2.upstream.DefaultHttpDataSource;

public class PlayerActivity extends AppCompatActivity {

    private ExoPlayer player;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player);

        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);

        String name = getIntent().getStringExtra("name");
        String url  = getIntent().getStringExtra("url");
        String logo = getIntent().getStringExtra("logo");

        TextView tvName       = findViewById(R.id.tvName);
        ImageView ivLogo      = findViewById(R.id.ivLogo);
        TextView tvLoading    = findViewById(R.id.tvLoading);
        ImageButton btnBack   = findViewById(R.id.btnBack);
        StyledPlayerView playerView = findViewById(R.id.playerView);

        tvName.setText(name);
        Glide.with(this).load(logo).into(ivLogo);
        btnBack.setOnClickListener(v -> finish());

        player = new ExoPlayer.Builder(this).build();
        playerView.setPlayer(player);

        HlsMediaSource mediaSource = new HlsMediaSource.Factory(
                new DefaultHttpDataSource.Factory())
                .createMediaSource(MediaItem.fromUri(url));

        player.setMediaSource(mediaSource);
        player.prepare();
        player.setPlayWhenReady(true);

        player.addListener(new Player.Listener() {
            @Override
            public void onPlaybackStateChanged(int state) {
                tvLoading.setVisibility(
                        state == Player.STATE_BUFFERING ? View.VISIBLE : View.GONE);
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (player != null) {
            player.release();
            player = null;
        }
    }
}

File: NayanTv/app/src/main/kotlin/com/iptv/nayantv/PlayerActivity.kt
package com.iptv.nayantv

import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.media3.common.MediaItem
import androidx.media3.common.Player
import androidx.media3.datasource.DefaultHttpDataSource
import androidx.media3.exoplayer.ExoPlayer
import androidx.media3.exoplayer.hls.HlsMediaSource
import androidx.media3.ui.PlayerView
import com.bumptech.glide.Glide

class PlayerActivity : AppCompatActivity() {

    private var player: ExoPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_player)

        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)

        val name = intent.getStringExtra("name") ?: ""
        val url  = intent.getStringExtra("url")  ?: ""
        val logo = intent.getStringExtra("logo") ?: ""

        val tvName    = findViewById<TextView>(R.id.tvName)
        val ivLogo    = findViewById<ImageView>(R.id.ivLogo)
        val tvLoading = findViewById<TextView>(R.id.tvLoading)
        val btnBack   = findViewById<ImageButton>(R.id.btnBack)
        val playerView = findViewById<PlayerView>(R.id.playerView)

        tvName.text = name
        Glide.with(this).load(logo).into(ivLogo)
        btnBack.setOnClickListener { finish() }

        player = ExoPlayer.Builder(this).build().also { exo ->
            playerView.player = exo
            val src = HlsMediaSource.Factory(DefaultHttpDataSource.Factory())
                .createMediaSource(MediaItem.fromUri(url))
            exo.setMediaSource(src)
            exo.prepare()
            exo.playWhenReady = true

            exo.addListener(object : Player.Listener {
                override fun onPlaybackStateChanged(state: Int) {
                    tvLoading.visibility =
                        if (state == Player.STATE_BUFFERING) View.VISIBLE else View.GONE
                }
            })
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        player?.release()
    }
}
